### PR TITLE
505 display time spents in the ticket timeline formatted as hours

### PR DESCRIPTION
**Pull request by sassani134**
## Related issue(s)

#505 
## How to test manually

- create a contract with a billing interval of 30 minutes
- attach the contract to a ticket
- answer to the ticket with spent time (e.g. 70 minutes)
- check the time is displayed as "1h 10m spent (1h 30m charged)"

## Checklist

- [x] code is manually tested
- [x] permissions / authorizations are verified
- [x] interface works on both mobiles and big screens
- [x] interface works in both light and dark modes
- [x] accessibility has been tested
- [x] tests are up-to-date
- [x] locales are synchronized
- [x] copyright notices are up to date
- [x] documentation is up to date (including migration notes)

### CODE REVIEW

File: translations/messages+intl-icu.en_GB.yaml [hunk 1]
```diff
@@ -102,9 +102,12 @@ home.no_tickets: 'No tickets'
 home.title: 'Welcome to Bileto'
 home.your_organizations: '{count, plural, one {Your organization} other {Your organizations}}'
 home.your_tickets: 'Your tickets'
-hours_formatter.hours: '{hours}h'
-hours_formatter.hours_and_minutes: "{hours}h\_{minutes}m"
-hours_formatter.minutes: '{minutes}m'
+hours_formatter.hours: '{hours} h'
+hours_formatter.hours_and_minutes: "{hours} h\_{minutes} m"
+hours_formatter.minutes: '{minutes} m'
```
Comment (marien-probesys) [inline translations/messages+intl-icu.en_GB.yaml:107]: Why did you add a space between the number and the unit?

File: translations/messages+intl-icu.en_GB.yaml [hunk 2]
```diff
@@ -102,9 +102,12 @@ home.no_tickets: 'No tickets'
...
+hours_formatter.hours.long: '{hours} hour'
+hours_formatter.hours_and_minutes.long: "{hours} hour\_{minutes} minute"
+hours_formatter.minutes.long: '{minutes} minute'
```
Comment (marien-probesys) [inline translations/messages+intl-icu.en_GB.yaml:110]: What if there are more than 1 hour/minute? This will display "2 hour 30 minute" for instance. You must use the pluralization formatter from ICU (some information here: https://formatjs.io/docs/core-concepts/icu-syntax/#plural-format). For instance:

```yaml
hours_formatter.hours.long: '{ hours, plural, one {1\_hour} other {#\_hours} }'
```

Note that I use the meta-character `\_` instead of a a simple space. `\_` is replaced by a non-breakable space. It means that if the string "1 hour" is placed at the end of a line, the unit will not be separated from the number. See the difference:

With a simple space:

"1 heure 30 minutes" is split on two lines, 1 and "heure" are separated

With a non-breakable space:

"1 heure 30 minutes" is put on a new line, 1 and "heure" are not separated

File: translations/messages+intl-icu.en_GB.yaml [hunk 3]
```diff
@@ -354,7 +357,7 @@ tickets.show.resolved: 'This ticket is resolved.'
 tickets.show.scroll_to_bottom: 'Scroll to bottom'
 tickets.show.show_history: 'Show history'
 tickets.show.time_spent: "{count, plural, one {1\_minute spent} other {#\_minutes spent}}"
-tickets.show.time_spent.accounted: "({count, plural, one {1\_minute accounted} other {#\_minutes accounted}})"
+tickets.show.time_spent.accounted: "({ count } accounted)"
```
Comment (marien-probesys) [inline translations/messages+intl-icu.en_GB.yaml:360]: :+1:

File: translations/messages+intl-icu.fr_FR.yaml [hunk 4]
```diff
@@ -105,6 +105,9 @@ home.your_tickets: 'Vos tickets'
 hours_formatter.hours: '{hours}h'
 hours_formatter.hours_and_minutes: "{hours}h\_{minutes}m"
 hours_formatter.minutes: '{minutes}m'
+hours_formatter.hours.long: '{hours} heure'
+hours_formatter.minutes.long: '{minutes} minutes'
+hours_formatter.hours_and_minutes.long: "{hours} heure\_{minutes} minutes"
```
Comment (marien-probesys) [inline translations/messages+intl-icu.fr_FR.yaml:110]: Same logic than above concerning the pluralization and spaces.

File: translations/messages+intl-icu.fr_FR.yaml [hunk 5]
```diff
@@ -353,8 +356,7 @@ tickets.show.refuse: Refuser
 tickets.show.resolved: 'Ce ticket est résolu.'
 tickets.show.scroll_to_bottom: 'Défiler en bas'
 tickets.show.show_history: 'Afficher l’historique'
-tickets.show.time_spent: "{count, plural, one {1\_minute passée} other {#\_minutes passées}}"
-tickets.show.time_spent.accounted: "({count, plural, one {1\_minute comptabilisée} other {#\_minutes comptabilisées}})"
+tickets.show.time_spent.accounted: "({ count } comptabilisé)"
```
Comment (marien-probesys) [inline translations/messages+intl-icu.fr_FR.yaml:359]:
```suggestion
tickets.show.time_spent.accounted: "({ count } comptabilisée(s))"
```

File: translations/messages+intl-icu.fr_FR.yaml [hunk 6]
```diff
@@ -425,4 +427,4 @@ users.no_organization_authorization: 'Cet utilisateur ne peut pas créer de tick
 users.no_organization: 'Aucune organisation'
 users.organization: Organisation
 users.password: 'Mot de passe'
-users.yourself: vous
+users.yourself: vous
```
Comment (marien-probesys) [inline translations/messages+intl-icu.fr_FR.yaml:430]: You removed the new line character at the end of this file. You must not. It is very often recommended (if not required) to keep it so when you open several files in the terminal (with `cat` for instance), the lines of each file are correctly displayed. You can learn more about it on StackOverflow: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

File: translations/messages+intl-icu.en_GB.yaml (hunk 1 above)
Comment (sassani134) [reply translations/messages+intl-icu.en_GB.yaml:107]: I thought the convention was to add a space between values ans the units sorry

File: src/Twig/HoursFormatterExtension.php [hunk 7]
```diff
@@ -42,29 +42,46 @@ public function formatHours(int|float $hours): string
     /**
      * Return the given minutes formatted as hours and minutes.
      */
-    public function formatMinutes(int $minutes): string
+    public function formatMinutes(int $minutes, bool $longFormat = true): string
```
[the remark this answers is not in the document]
Comment (sassani134) [reply src/Twig/HoursFormatterExtension.php]: done, i add the same option here:
[a16861c](https://github.com/Probesys/bileto/pull/577/commits/a16861ce29a368c4335b291dc71dadbeb2b76d46)

**Comment by sassani134** [comment]
Sorry i mistakenly named my two first commit Update src/Twig/HoursFormatterExtension.php

File: translations/messages+intl-icu.en_GB.yaml (hunk 1 above)
Comment (marien-probesys) [reply translations/messages+intl-icu.en_GB.yaml]: Your argument is good, don't worry :P That was my initial mistake!

Note that we should use non-breaking spaces so the number is not separated of its unit at the end of a line (see the figure 1 in this article https://intelligentediting.com/blog/non-breaking-spaces-how-to-check-something-you-cannot-even-see/ for instance). As we cannot distinguish a normal space from a non-breaking space, we can use the special character `\_` in YAML to represent a non-breaking space.

**Comment by marien-probesys** [comment]
> Sorry i mistakenly named my two first commit Update src/Twig/HoursFormatterExtension.php

I wouldn't keep all your commits anyway in order to keep the Git history clean: I merged all of them into a single commit.

Note that I also fixed the latest remarks, and I changed the boolean parameter `$longFormat` in a string parameter `$format` (valid values are "short" and "long"). This makes the function calls a little easier to understand: `formatMinutes(true)` vs. `formatMinutes('long')`.

I rebased all your work on the `main` branch, and I'm going to merge it :tada:

**Review by marien-probesys — APPROVED** [review]
